### DISCUSSION TITLE
Allow environment variables in astro.config.* files

### DISCUSSION THREAD

**Discussion by susickypavel** [withastro/roadmap#269] [Proposal]
Hi 👋

I think it is quite frequent to use environment variables inside config files to change compiling/bundling/transpiling behavior per mode (development, production, ...)

It's also weird that this isn't supported by itself because of Vite.js. Overall env var behavior feels a little bit out of place since `NODE_ENV` is undefined in dev/build.

**Comment by jkjustjoshing** [comment]
This is the only reason I have to use env vars.

I have a CI system that deploys each PR branch to a subdirectory of our single staging environment. So I need to update the `base` value with an environment variable during the build process. If I can't access env vars in astro.config.*, then I can't update the `base` each time I run the build.

**Comment by jkjustjoshing** [comment]
As a workaround, you can install the `dotenv` package and make the following changes to your package.json file:
```diff
-  "dev": "astro dev",
-  "build": "astro build",
+  "dev": "node -r dotenv/config ./node_modules/.bin/astro dev",
+  "build": "node -r dotenv/config ./node_modules/.bin/astro build",
```

This forces your `.env` values to be loaded before astro starts. Use `process.env` instead of `import.meta.env`.

Are there any downsides to this method that I'm not thinking of?
